R/write_pxl_file.R

```diff
@@ -0,0 +1,499 @@
+#'  filtered to include the components currently available in the \code{Seurat} object,
+#'  and then the component IDs are updated. The resulting merged edgelist data is exported
+#'  to a .parquet file.
+#'  - Sample meta data : The sample meta data is extracted from the original .pxl file(s),
+#'  then merged and exported to a .json file.
+#'
+#' The structure of the .pxl file is detailed below:
+#'
+#' |-- adata.h5ad\cr
+#' |-- polarization.parquet\cr
+#' |-- colocalization.parquet\cr
+#' |-- metadata.json\cr
+#' |-- edgelist.parquet
+#'
+#' The merged files are converted into a zip archive and saved to the target .pxl file.
+#'
+#' @param object A \code{Seurat} object with a \code{CellGraphAssay5}
+#' assay object created with pixelatorR.
+#' @param file A character string specifying the path to the .pxl
+#' file to be created.
+#' @param assay A character string specifying the name of the \code{CellGraphAssay5}.
+#' If set to \code{NULL} the default assay will be used.
+#' @param overwrite A logical value specifying whether to overwrite the \code{file}
+#' if it already exists.
+#'
+#' @return Nothing. The function writes the .pxl file to the specified location.
+#'
+#' @examples
+#' # Use Assay5 as the default assay version
+#' options(Seurat.object.assay.version = "v5")
+#'
+#' # Create Seurat object
+#' pxl_file <- system.file("extdata/five_cells",
+#'                         "five_cells.pxl",
+#'                         package = "pixelatorR")
+#' se <- ReadMPX_Seurat(pxl_file)
+#'
+#' se_merged <- merge(se, list(se, se, se))
+#' pxl_file <- fs::path_temp("small.pxl")
+#'
+#' # Export data to a new .pxl file
+#' WriteMPX_pxl_file(se_merged, pxl_file)
+#'
+#' # Read the new .pxl file
+#' se_merged <- ReadMPX_Seurat(pxl_file)
+#'
+#' # Reset global option
+#' options(Seurat.object.assay.version = "v3")
+#'
+#' @export
+#'
+WriteMPX_pxl_file <- function (
+  object,
+  file,
+  assay = NULL,
+  overwrite = FALSE
+) {
+
+  stopifnot(
+    "'object' must be a Seurat object" =
+      inherits(object, what = "Seurat"),
+    "'file' must be a valid path ending with .pxl" =
+      is.character(file) && (fs::path_ext(file) == "pxl"),
+    "'overwrite' must be either TRUE or FALSE" =
+      is.logical(overwrite)
+  )
+
+  # Check if file exists
+  if (fs::file_exists(file) & !overwrite) {
+    abort(glue("{col_br_blue(file)} already exists. Please select a different ",
+               "file name or set overwrite = TRUE if you are certain that the ",
+               "existing file should be replaced."))
+  }
+
+  assay <- assay %||% DefaultAssay(object)
+  cg_assay <- object[[assay]]
+  stopifnot(
+    "'assay' must be a 'CellGraphAssay' or a 'CellGraphAssay5' object" =
+      is(cg_assay, "MPXAssay")
+  )
+
+  # fetch and validate fs_map
+  fs_map <- cg_assay@fs_map
+  .validate_fs_map(fs_map)
+
+  # Delete the file only if it's not in use
+  # by the current object
+  if (fs::file_exists(file)) {
```

Comment:
If the file exists, 
check if file is in the file system map
If it is, cancel the file creation
Otherwise, remove the file

Why is the fiile removed? Is it because it should be overwritten and replaced?

Reply:
Yeah you're right. It doesn't have to be deleted to proceed. I added it as an extra precaution in case the file is write-protected.

Reply:
precautions are good! Let's keep it